check_diff/src/lib.rs

```diff
@@ -56,3 +162,146 @@ pub fn change_directory_to_path(dest: &Path) -> io::Result<()> {
     );
     return Ok(());
 }
+
+pub fn get_ld_lib_path() -> Result<String, CheckDiffError> {
+    let Ok(command) = Command::new("rustc").args(["--print", "sysroot"]).output() else {
+        return Err(CheckDiffError::FailedCommand(
+            "Error getting sysroot".to_string(),
+        ));
```

Comment:
As I mentioned above we probably don't ned to store `Strings` in the `CheckDiffError` error enum. In this case we could just store a ``&`static str`` instead of allocating a String. The same seems to apply to `FailedUtf8`, `FailedCargoVersion`, `FailedSourceBuild`